storage/storage_helpers.cpp

```diff
@@ -20,20 +20,25 @@ bool IsDownloadFailed(Status status)
          status == Status::EUnknown;
 }
 
-bool IsEnoughSpaceForDownload(TMwmSize size)
+bool IsEnoughSpaceForDownload(TMwmSize mwmSize, TMwmSize maxMwmSize)
 {
-  // Mwm size is less than kMaxMwmSizeBytes. In case of map update at first we download updated map
+  // Mwm size is less than |maxMwmSize|. In case of map update at first we download updated map
   // and only after that we do delete the obsolete map. So in such a case we might need up to
   // kMaxMwmSizeBytes of extra space.
-  return GetPlatform().GetWritableStorageStatus(size + kMaxMwmSizeBytes) ==
+  return GetPlatform().GetWritableStorageStatus(mwmSize + maxMwmSize) ==
          Platform::TStorageStatus::STORAGE_OK;
 }
 
 bool IsEnoughSpaceForDownload(TCountryId const & countryId, Storage const & storage)
 {
   NodeAttrs nodeAttrs;
   storage.GetNodeAttrs(countryId, nodeAttrs);
-  return IsEnoughSpaceForDownload(nodeAttrs.m_mwmSize - nodeAttrs.m_localMwmSize);
+  // The type of nodeAttrs.m_mwmSize and nodeAttrs.m_localMwmSize is TMwmSize (uint64_t).
```

Comment:
Our question was not about integer overflow, but about reasons why local mwm size may be less than the remote mwm size. I suggest you following comment:
```
Size of local mwm may be greater than size of remote mwm by some reason (new compression algorithm, removal of unused sections, etc).
```